lib/mayaUsd/ufe/UsdAttribute.cpp

```diff
@@ -81,8 +81,7 @@ template <typename T> bool setUsdAttr(const PXR_NS::UsdAttribute& attr, const T&
     std::string                                     errMsg;
     bool isSetAttrAllowed = MayaUsd::ufe::isAttributeEditAllowed(attr, &errMsg);
     if (!isSetAttrAllowed) {
-        MGlobal::displayError(errMsg.c_str());
-        return false;
+        throw std::runtime_error(errMsg);
```

Comment:
If you're not catching this exception in this file, it will escape to calling code, which is possibly not prepared to deal with it.  This feels like quite an important change, with risk.  I would like to understand if the benefits outweigh the costs.

Reply:
We had a discussion before where I pointed out the benefit of not letting code run when the attribute it intended to set was not really set. No unit tests were broken with this change and it allows code that want to detect incorrect attribute change to do so. This is taken advantage in new Maya code in the attribute editor, for example. (Arguably, there were already possibility of exceptions, like out of memory, but much more rarely.)